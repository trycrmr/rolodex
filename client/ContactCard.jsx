import React from "react";
import { Card, Media, Heading, Content, Image } from "react-bulma-components";
import { Link } from "react-router-dom";

const ContactCard = (props) => {
  return (
    <>
      <Card>
        <Card.Header></Card.Header>
        <Card.Content>
          <Media position="right">
            <Media.Item renderAs="figure" position="left">
              <Image
                size={128}
                alt="128x128"
                src="http://bulma.io/images/placeholders/128x128.png"
              />
            </Media.Item>
            <Media.Item style={{ margin: "auto", textAlign: "center" }}>
              <Heading size={6}>{props.data.name}</Heading>
              <Content>
                {props.data.email}
                <br />
                {props.data.phone}
              </Content>
            </Media.Item>
          </Media>
        </Card.Content>
        <Card.Footer>
          <Card.Footer.Item renderAs="div" href="/view">
            <Link to={`/view/${props.data.id}`}>View</Link>
          </Card.Footer.Item>
          <Card.Footer.Item renderAs="div" href="/edit">
            <Link to={`/edit/${props.data.id}`}>Edit</Link>
          </Card.Footer.Item>
          <Card.Footer.Item
            renderAs="a"
            onClick={() =>
              alert(
                `Are you sure you want to remove ${props.data.name} from your contacts?`
              )
            }
          >
            Delete
          </Card.Footer.Item>
        </Card.Footer>
      </Card>
    </>
  );
};

export default ContactCard;
