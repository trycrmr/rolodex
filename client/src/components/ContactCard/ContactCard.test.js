import React from "react";
import { mount } from "enzyme";
import ContactCard from "./index";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

const fakeContact = {
  name: "Milan Jenkins",
  email: "Kristian_Deckow@example.org",
  phone: "330-013-0169",
  id: "e12970f9-1deb-48a2-a4c2-854d86a82ef0",
};

describe("The Contact Card component", () => {
  it("renders without crashing", () => {
    mount(
      <MemoryRouter>
        <ContactCard data={fakeContact} />
      </MemoryRouter>
    );
  });

  it("should match snapshot", () => {
    // validates fields and links render as expected given the dataset
    expect(
      renderer
        .create(
          <MemoryRouter>
            <ContactCard data={fakeContact} />
          </MemoryRouter>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
