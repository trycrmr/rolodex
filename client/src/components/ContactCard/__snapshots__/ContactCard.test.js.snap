// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Contact Card component should match snapshot 1`] = `
<div
  className="card"
  style={
    Object {
      "margin": "auto",
      "maxWidth": "500px",
    }
  }
>
  <div
    className="card-content"
  >
    <article
      className="media"
      position="right"
    >
      <figure
        className="media-left"
      >
        <figure
          className="image is-128x128"
        >
          <img
            alt="128x128"
            className=""
            onError={[Function]}
            src="http://bulma.io/images/placeholders/128x128.png"
          />
        </figure>
      </figure>
      <div
        className="media-content"
        style={
          Object {
            "margin": "auto",
            "textAlign": "center",
          }
        }
      >
        <h1
          className="title is-6"
        >
          Milan Jenkins
        </h1>
        <div
          className="content"
        >
          Kristian_Deckow@example.org
          <br />
          330-013-0169
        </div>
      </div>
    </article>
  </div>
  <div
    className="card-footer"
  >
    <div
      className="card-footer-item"
      href="/view"
    >
      <a
        href="/view/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
        onClick={[Function]}
      >
        View
      </a>
    </div>
    <div
      className="card-footer-item"
      href="/edit"
    >
      <a
        href="/edit/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
        onClick={[Function]}
      >
        Edit
      </a>
    </div>
    <div
      className="card-footer-item"
      href="/delete"
    >
      <a
        href="/delete/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
        onClick={[Function]}
      >
        Delete
      </a>
    </div>
  </div>
</div>
`;
